### PR TITLE
Add patch info to list cmd

### PR DESCRIPTION
**Pull request by jgeudens**
This PR adds the patch info to the metadata so it can be shown  by the list command. This was logged as #198.

I have added a dedicated feature test, because otherwise it would make an existing test more complex.

### CODE REVIEW

**Review by ben-edna — APPROVED** [review]
Looks good!

**Comment by jgeudens** [comment]
Thanks,

There is a pylint warning about which I'm not sure how to fix.

**Comment by ben-edna** [comment]
> Thanks,
> 
> There is a pylint warning about which I'm not sure how to fix.

Maybe if we replace self._tag, self._branch and self._revision with a self._version of type Version, this gives enough room for the new self._patch?

File: dfetch/project/metadata.py [hunk 1]
```diff
@@ -110,9 +114,14 @@ def last_fetch_string(self) -> str:
 
     @property
     def hash(self) -> str:
-        """Hash of directory."""
+        """Hash of directory as stored in the metadata.."""
         return self._hash
 
+    @property
+    def patch(self) -> str:
+        """Applied patch as stored in the metadata.."""
```
Comment (ben-edna) [inline dfetch/project/metadata.py:122]:
```suggestion
        """The applied patch as stored in the metadata."""
```

**Review by ben-edna — CHANGES_REQUESTED** [review]

**Comment by jgeudens** [comment]
Pylint warning has been fixed.

File: dfetch/project/metadata.py (hunk 1 above)
Comment (jgeudens) [reply dfetch/project/metadata.py]: Applied

**Pull request merged**